app/controllers/api/messages_controller.rb

```diff
@@ -0,0 +1,154 @@
+# The MessagesController is the RESTful interface to Message objects
+
+module Api
+  class MessagesController < ApiController
+    before_action :authorize
+
+    before_action :check_api_writable, :only => [:create, :update, :destroy]
+    before_action :check_api_readable, :except => [:create, :update, :destroy]
```

Comment:
This should be `inbox`, `outbox` and `show` surely?

Reply:
@tomhughes What are you suggesting here - `inbox`, `outbox` and `show` are covered by `:check_api_readable`, did you mean to exclude them as well?

Reply:
Oh sorry I hadn't noticed that was one `:except` and was reading them both as `:only`.